ExploreASL_Initialize.m

```diff
@@ -697,111 +639,242 @@
+            warning(['x.' field_symbol{iField} ' was not defined in DATA_PAR.m!'])
+        end
+    end
+
+    if ~isfield(x,'D')
+        warning('x.D didn''nt exist');
+    else
+        field_symbol = {'ROOT'};
+        for iField=1:length(field_symbol)
+            if ~isfield(x.D,field_symbol{iField})
+                warning(['x.D.' field_symbol{iField} ' was not defined in DATA_PAR.m!'])
             end
-            if      x.S.Sets1_2Sample(iSet)==1
-                    fprintf(', codes for paired data');
-            elseif  x.S.Sets1_2Sample(iSet)==2
-                    fprintf(', codes for two-sample data');
-            elseif  x.S.Sets1_2Sample(iSet)==3
-                    fprintf([', continuous variate (with ' num2str(length(unique(x.S.SetsID(:,iSet)))) ' unique values)']);
-            end                    
-
-            fprintf('\n');
         end
-else    
-    fprintf('%s\n','No sets are defined');
-end
+    end
+
+    if ~isempty(regexp(x.subject_regexp, '^(\^|)\.\*(\$|)$'))
+        warning('Subject regexp not specific! Check that no wrong folders are included as subjects');
+    end
+
+    fprintf('\n');
 
-if ~isfield(x,'M0')
-%     warning('M0 option missing!');
-else
-    fprintf('\n%s\n',['M0 option selected is "' num2str(x.M0) '"']);
 end
 
-fprintf('%s\n',['x.DELETETEMP = ' num2str(x.DELETETEMP) ' (delete temporary files)']);
-fprintf('%s\n',['x.Quality    = ' num2str(x.Quality) ' (0 = fast try-out; 1 = normal high quality)']);
+%% -----------------------------------------------------------------------
+%% Add ExploreASL Directory
+function addExploreASLDirectory(MyPath)
 
+    % Define paths (should be equal when loading data or only initializing)
+    addpath(MyPath); % ExploreASL
+    subfolders_to_add = {'Functions', 'mex', 'Modules', 'Testing',...
+        fullfile('Modules', 'SubModule_Structural'), ...
+        fullfile('Modules', 'SubModule_ASL'), ...
+        fullfile('Modules', 'SubModule_Population'), ...
+        'Development', 'External', ...
+        fullfile('External','isnear'), ...
+        fullfile('External','DCMTK'), ...
+        fullfile('Testing', 'UnitTests'), ...
+        fullfile('External','ExploreQC'), ...
+        fullfile('External','SPMmodified'), ...
+        fullfile('External','SPMmodified','matlabbatch'),...
+        fullfile('External','SPMmodified','xASL'),...
+        fullfile('External','SPMmodified','toolbox','cat12'), ...
+        fullfile('External','SPMmodified','toolbox','LST'), ...
+        fullfile('External','SPMmodified','toolbox','OldNorm') ...
+        genpath(fullfile('External','bids-matlab'))};
+    % Iterate over subfolders which should be added
+    for ii=1:length(subfolders_to_add)
+        addpath(fullfile(MyPath,subfolders_to_add{ii}));
+    end
+    
+end
 
 %% -----------------------------------------------------------------------
-%% 3) Print warnings
-fprintf('\n%s\n\n','---------------------------------------------');
+%% Define input parser
```

Comment:
What is the advantage now of this, if the code only becomes longer with this way of input parsing? ;)
Also, it makes sense to read the input parsing at the first part of the code right. Though it's nice that it's separated here below in a subfunction, it becomes more difficult to debug.

Reply:
Length isn't everything ;)

I think the benefit is that it's modularized now. Of course we could write it in a more simplified way like before, but the use of varargin and the input parser are the most high level solutions described in here: https://docs.google.com/document/d/1M9GTXbRe1VngH9M3p6jyVABqtFgnSYDgpL13MnCelHQ/edit?usp=sharing 

And instead of multiple if-else-statements, you just change the default value or the limiting function handle instead. That seems a lot more sophisticated to me. Why should debugging be harder now?

Reply:
That's true. I just thought of debugging somewhere in the script, and usually a script then more or less has a top-down order, where you see at the top which parameters are initialized where and why. So it is faster/easier to see how this works. Now, you have to search for the input parsing function, realize that this is a subfunction, search for the subfunction, and check what's going on there. 

It's also nice to have in 1 screen the function call, header, input parsing, and admin, rather than having the input parsing at the end of the script. Perhaps just intuitively weird. But practice probably will teach me otherwise ;)